recommender-back/src/apis/poi.py

```diff
@@ -72,12 +74,20 @@ def calculate_score(self, cur_time=None, sunrise=None, sunset=None):
                     scorer = self.scorers["Indoor"]
 
                 if scorer:
-                    data['Score'] = scorer.score(
-                        weather_data['temperature'], weather_data['wind_speed'],
-                        weather_data['humidity'], weather_data['precipitation'],
-                        weather_data['clouds'], weather_data['air_quality'], 
-                        sunrise_time, sunset_time, current_time
-                    )
+                    if weather_data.get('air_quality'):
+                        data['Score'] = scorer.score(
+                            weather_data['temperature'], weather_data['wind_speed'],
+                            weather_data['humidity'], weather_data['precipitation'],
+                            weather_data['clouds'], weather_data['air_quality'], 
+                            sunrise_time, sunset_time, current_time
+                        )
+                    else:
+                        data['Score'] = scorer.score(
+                            weather_data['temperature'], weather_data['wind_speed'],
+                            weather_data['humidity'], weather_data['precipitation'],
+                            weather_data['clouds'], 0,
+                            sunrise_time, sunset_time, current_time
+                        )
```

Comment:
There's some duplicate code here that could be refactored.
For example:
`temperature = weather_data.get('temperature')
wind_speed = weather_data.get('wind_speed')
humidity = weather_data.get('humidity')
precipitation = weather_data.get('precipitation')
clouds = weather_data.get('clouds')
air_quality = weather_data.get('air_quality', 0)

data['Score'] = scorer.score(
    temperature, wind_speed, humidity, precipitation, clouds, air_quality,
    sunrise_time, sunset_time, current_time
)`